Week1/homework/index.js

```diff
@@ -0,0 +1,205 @@
+  connection.query(createCountries, (error, results, fields) => {
+    if (error) {
+      console.log(error);
+      res.send(`Table hasn't created ${error}`);
+      connection.end();
+    }
+    console.log(`Table created`);
+    res.send(`Table created`);
+  });
+});
+
+app.put('/createCities', (req, res) => {
+  connection.query(createCities, (error, results, fields) => {
+    if (error) {
+      console.log(error);
+      res.send(`Table hasn't created ${error}`);
+      connection.end();
+    }
+    console.log(`Table created`);
+    res.send(`Table created`);
+  });
+});
+
+app.post('/appendCountries', async (req, res) => {
+  try {
+    const { countries } = data;
+    for (const country of countries) {
+      const try1 = `INSERT INTO countries VALUES('${country.name}', '${country.continent}', '${country.region}', '${country.surface_area}', ' ${country.indep_year}', ' ${country.population}', ' ${country.life_expectancy}', ' ${country.gnp}', ' ${country.gnp_old}', ' ${country.local_name}', ' ${country.government_form}', ' ${country.head_of_state}', ' ${country.capital}');`;
+      await connection.query(try1, (error, results, fields) => {
+        if (error) {
+          console.log(`Table hasn't values ${error}`);
+        }
+      });
+    }
+    console.log('Table has values ');
+    res.write(`Success `);
+  } catch (err) {
+    res.write(`Error: ${err}`);
+  }
+  res.write(`Table has values`);
+  res.end();
+});
+
+app.post('/appendCities', async (req, res) => {
+  try {
+    const { cities } = data;
+    for (const city of cities) {
+      const try1 = `INSERT INTO cities values(NULL, '${city.name}', '${city.country_code}', '${city.district}', ' ${city.population}'); `;
+      await connection.query(try1, (error, results, fields) => {
+        if (error) {
+          console.log(`Table hasn't values ${error}`);
+          res.write(`Table hasn't values ${error}`);
+          res.end();
+        }
+      });
+    }
+    console.log('Table has values ');
+    res.write(`Success `);
+  } catch (err) {
+    res.write(err);
+  }
+  res.write(`Table has values`);
+  res.end();
+});
+
+app.get('/greaterThan8m', (req, res) => {
+  connection.query(
+    'SELECT * FROM countries WHERE Population >= 8000000;',
+    (error, results, fields) => {
+      if (error) {
+        console.log(error);
+        connection.end();
+      }
+      console.log(`Cities logged`);
+      res.send(results);
+    },
+  );
+});
+
+app.get('/haveLandName', (req, res) => {
+  connection.query(
+    `SELECT * FROM countries WHERE countries.Name like "%land%" ;`,
+    (error, results, fields) => {
+      if (error) {
+        console.log(error);
+        connection.end();
+      } else {
+        console.log(`Cities name with "land"`);
+        res.send(results);
+      }
+    },
+  );
+});
+
+app.get('/cityPopulation', (req, res) => {
+  connection.query(
+    `SELECT * FROM cities WHERE Population >500000 and Population <1000000;`,
```

Comment:
If you want you can also use between here

Reply:
got it 👍